Panic when starting service with no logger

## Is your feature request related to a problem? Please describe

No.

## Is this a bug? Please provide steps to reproduce, a failing test etc.

With the introduction of a new logger functionality in #259 I noticed some of the examples panicking. 

In the release notes we say that
> Keep in mind that if no logger is provided it will use the default structured logger

But it seems that services that are set up without a logger instead panic.

Right now, examples `second`, `fourth` and `fifth` panic for this exact cause.

Here's what happens for me locally in the examples, or if I run `service, err := patron.New(name, version)`

```
$ go run fourth/main.go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x164234b]

goroutine 1 [running]:
github.com/rs/zerolog.(*Logger).should(0x0, 0x16da301, 0xc)
	/Users/paschalistsilias/go/pkg/mod/github.com/rs/zerolog@v1.5.0/log.go:358 +0x2b
github.com/rs/zerolog.(*Logger).newEvent(0x0, 0x1010101, 0x0, 0x16da380)
	/Users/paschalistsilias/go/pkg/mod/github.com/rs/zerolog@v1.5.0/log.go:337 +0x3e
github.com/rs/zerolog.(*Logger).Info(...)
	/Users/paschalistsilias/go/pkg/mod/github.com/rs/zerolog@v1.5.0/log.go:241
github.com/beatlabs/patron/log/zerolog.(*Logger).Infof(0xc00000eda0, 0x17f09aa, 0x2f, 0xc00007e3f0, 0x3, 0x3)
	/Users/paschalistsilias/Beat-repos/patron/log/zerolog/logger.go:108 +0x3d
github.com/beatlabs/patron/log.Infof(...)
	/Users/paschalistsilias/Beat-repos/patron/log/log.go:150
github.com/beatlabs/patron.setupTracing(0x17d01eb, 0x6, 0x17cebfb, 0x5, 0x0, 0xc00007e390)
	/Users/paschalistsilias/Beat-repos/patron/service.go:305 +0x2fa
github.com/beatlabs/patron.setupObservability(0x17d01eb, 0x6, 0x17cebfb, 0x5, 0xc00007e390, 0x18d4e20, 0xc00000ed60, 0xc00002a0b0, 0xc00010fe68)
	/Users/paschalistsilias/Beat-repos/patron/service.go:271 +0x97
github.com/beatlabs/patron.New(0x17d01eb, 0x6, 0x17cebfb, 0x5, 0x0, 0x0, 0x0, 0x1cb2cf0, 0x1801cc0, 0xc00010feb8)
	/Users/paschalistsilias/Beat-repos/patron/service.go:228 +0x1e8
main.main()
	/Users/paschalistsilias/Beat-repos/patron/examples/fourth/main.go:71 +0x71
exit status 2
```

The panic starts on [this](https://github.com/beatlabs/patron/blob/3ba61f44574263cf2094a6e98b3232324cf3eb3b/service.go#L305) line, where we log something on a nil logger.

## Describe the solution

As we've discussed, we can keep the original idea of falling back to the default structured logger if no other option is provided.

Another idea would be to have a dummy no-op logger, but I'm not a big fan to be honest.

## Additional context

None, for now

--- a/examples/fifth/main.go
+++ b/examples/fifth/main.go
@@ -58,7 +58,7 @@ func main() {
 		os.Exit(1)
 	}
 
-	cc, err := patrongrpc.Dial("localhost:50006", grpc.WithInsecure(), grpc.WithBlock())
+	cc, err := patrongrpc.Dial("localhost:50006", grpc.WithInsecure())
 	if err != nil {
 		log.Fatalf("failed to dial grpc connection: %v", err)
 	}

--- a/log/zerolog/logger.go
+++ b/log/zerolog/logger.go
@@ -54,8 +54,9 @@ func (l *Logger) Sub(ff map[string]interface{}) log.Logger {
 	if ff == nil {
 		return l
 	}
-	sl := l.logger.With().Fields(ff).Logger()
-	return &Logger{logger: &sl, level: l.level}
+	logger := l.logger.With().Fields(ff).Logger()
+	loggerf := l.loggerf.With().Fields(ff).Logger()
+	return &Logger{logger: &logger, loggerf: &loggerf, level: l.level}
 }
 
 // Panic logging.

--- a/log/zerolog/logger_test.go
+++ b/log/zerolog/logger_test.go
@@ -30,7 +30,10 @@ func TestNewLogger(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			assert.NotNil(t, New(&zerolog.Logger{}, tt.lvl, tt.f))
+			sl := New(&zerolog.Logger{}, tt.lvl, tt.f)
+			assert.NotNil(t, sl)
+			assert.NotNil(t, sl.(*Logger).loggerf)
+			assert.NotNil(t, sl.(*Logger).logger)
 		})
 	}
 }
@@ -40,6 +43,9 @@ func TestLogger_Sub(t *testing.T) {
 	l := New(&b, log.DebugLevel, f)
 	sl := l.Sub(map[string]interface{}{"subkey1": "subval1"})
 	assert.NotNil(t, sl)
+	assert.NotNil(t, sl.(*Logger).loggerf)
+	assert.NotNil(t, sl.(*Logger).logger)
+
 	sl.Debug(logMsg)
 	assertLog(t, b, log.DebugLevel, logMsg)
 	assert.Contains(t, b.String(), `"subkey1":"subval1"`, b.String())

